src/chainstate/stacks/db/blocks.rs

```diff
@@ -5272,50 +5281,46 @@ impl StacksChainState {
             _ => false, // unused
         };
 
-        let current_tip =
-            StacksChainState::get_parent_index_block(current_consensus_hash, current_block);
-        let res = match self.with_read_only_clarity_tx(&NULL_BURN_STATE_DB, &current_tip, |conn| {
+        let current_tip = if use_unconfirmed_tip
+            && self.unconfirmed_state.is_some()
+            && self.unconfirmed_state.as_ref().unwrap().is_readable()
+        {
```

Comment:
Actually, I almost wonder if this code path here is actually fixing a bug in `StacksChainState::maybe_read_only_clarity_tx()` -- perhaps this method should instead be checking to see if the unconfirmed state trie actually exists?